core-bundle/src/Resources/contao/controllers/BackendPassword.php

```diff
@@ -52,25 +51,35 @@ public function __construct()
 	 */
 	public function run()
 	{
+		$container = System::getContainer();
+
 		/** @var Request $request */
-		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
+		$request = $container->get('request_stack')->getCurrentRequest();
+
+		Controller::loadDataContainer('tl_user');
+
+		$dc = new DC_Table('tl_user');
+		$dc->id = $this->User->id;
+		$dc->activeRecord = $this->User;
+
+		$widget = new Password(Password::getAttributesFromDca($GLOBALS['TL_DCA']['tl_user']['fields']['password'], 'password'));
+		$widget->template = 'be_widget_chpw';
+		$widget->dataContainer = $dc;
+		$widget->password = $GLOBALS['TL_LANG']['MSC']['password'][0];
+		$widget->confirm = $GLOBALS['TL_LANG']['MSC']['confirm'][0];
 
 		$objTemplate = new BackendTemplate('be_password');
+		$objTemplate->widget = $widget->parse();
 
 		if (Input::post('FORM_SUBMIT') == 'tl_password')
 		{
+			$widget->validate();
+
 			$pw = $request->request->get('password');
```

Comment:
@fritzmg Why are we not using `$widget->value` here instead of reading the POST data?

Reply:
No particular reason (other than this being untouched from before).

Reply:
Turns out we cannot use `$widget->value` because it returns the password hash instead of the actual password.